### PR TITLE
new entry java deque

### LINKED ISSUE #4451: [Concept Entry] Java Deque Deque
### Reason/inspiration (optional)

"We would like a new concept entry for [Java](https://www.codecademy.com/resources/docs/java) on Deque.
Provide the entry in a markdown file under `docs/content/java/concepts/deque/deque.md`

The entry should
    - Have an introductory paragraph about the deque data structure.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the use of deque.
    - Explain the equivalent method for stacks and queue.
Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by zottima**
### Description

New concept entry for Java: Deque

### Issue Solved

Closes #4451

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/java/concepts/deque/deque.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+[Cabbage, Sausage, Salad]
+```
+
+## Methods
+
+The following list contains a selection of methods provided by the `Deque` interface:
+
+- `.addFirst(item)`: Adds `item` at the top of the `Deque` if possible, otherwise it throws an exception.
+- `.addLast(item)`: Adds `item` at the end of the `Deque` if possible, otherwise it throws an exception.
+- `.getFirst()`: Returns, without removal, the first item of the `Deque`.
```
Comment (avdhoottt) [inline content/java/concepts/deque/deque.md:69]:
```suggestion
- `.getFirst()`: Returns, without removal, the first item of the `Deque`. It throws an exception if the Deque is empty.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey Author, I've suggested some changes, please make them. Thanks!!

File: content/java/concepts/deque/deque.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Syntax
+
+```pseudo
+import java.util.deque;
+
+Deque<DataType> myDeque = new DequeClass<DataType>();
+```
+
+Where [`DataType`](https://www.codecademy.com/resources/docs/java/data-types) is the data type to be stored in the deque, and `DequeClass` is a class implementing the `Deque` interface, for example, `ArrayDeque` or `LinkedList`.
```
[the remark this answers is not in the document]
Comment (zottima) [reply content/java/concepts/deque/deque.md:29]: Can you please elaborate? I don't think I understand what you mean.

**Comment by avdhoottt** [comment]
Hey @zottima, its been two weeks since I suggested the changes. Are you still working on this PR?

File: content/java/concepts/deque/deque.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+| `.addFirst(item)`  |             -             |       `.push(item)`       |
+|  `.addLast(item)`  |       `.add(item)`        |             -             |
+|   `.getFirst()`    |       `.element()`        |             -             |
+| `.offerLast(item)` |      `.offer(item)`       |             -             |
+|   `.peekFirst()`   |         `.peek()`         |         `.peek()`         |
+|   `.pollFirst()`   |         `.poll()`         |             -             |
+|  `.removeFirst()`  |        `.remove()`        |         `.pop()`          |
+
+
+
```
Comment (avdhoottt) [inline content/java/concepts/deque/deque.md:88]:
```suggestion
```

**Review by avdhoottt — APPROVED** [review]
LGTM!

**Comment by zottima** [comment]
> Hey @zottima, its been two weeks since I suggested the changes. Are you still working on this PR?

Yeah, I've been waiting for your reply to my question regarding one of your comments.

**Comment by avdhoottt** [comment]
> > Hey @zottima, its been two weeks since I suggested the changes. Are you still working on this PR?
> 
> Yeah, I've been waiting for your reply to my question regarding one of your comments.

Yes, you can ask it but I've already made the changes.

**Pull request merged**